query-algebrizer/src/types.rs

```diff
@@ -461,4 +464,140 @@ impl Debug for EmptyBecause {
             },
         }
     }
+}
+
+/// A set of `ValueType`s.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum ValueTypeSet {
+    None,
+    Any,
+    One(ValueType),
+    Many(EnumSet<ValueType>),
```

Comment:
But I'm OK with this; once the bugs get hammered out, and the assumptions documented, it will basically never be touched.  Consider making it `DistinguishedSet<T>` or something, since there's no dependence on `ValueType`.